sympy/logic/boolalg.py

```diff
@@ -498,6 +498,107 @@ def _eval_derivative(self, x):
+            changed = False
+            # Sort based on ordered
+            Rel = list(ordered(Rel))
+            # Create a list of possible replacements
+            results = []
+            # Try all combinations
+            for ((i, pi), (j, pj)) in combinations(enumerate(Rel), 2):
+                for k, (pattern, simp) in enumerate(patterns):
+                    res = []
+                    # use SymPy matching
+                    oldexpr = rv.func(pi, pj)
+                    tmpres = oldexpr.match(pattern)
+                    if tmpres:
+                        res.append((tmpres, oldexpr))
+                    # Try reversing first relational
+                    # This and the rest should not be required with a better canonical
+                    oldexpr = rv.func(pi.reversed, pj)
+                    tmpres = oldexpr.match(pattern)
+                    if tmpres:
+                        res.append((tmpres, oldexpr))
+                    # Try reversing second relational
+                    oldexpr = rv.func(pi, pj.reversed)
+                    tmpres = oldexpr.match(pattern)
+                    if tmpres:
+                        res.append((tmpres, oldexpr))
+                    # Try reversing both relationals
+                    oldexpr = rv.func(pi.reversed, pj.reversed)
+                    tmpres = oldexpr.match(pattern)
+                    if tmpres:
+                        res.append((tmpres, oldexpr))
+
+                    if res:
+                        for tmpres, oldexpr in res:
+                            # we have a matching, compute replacement
+                            np = simp.subs(tmpres)
+                            if np == dominatingvalue:
+                                # if dominatingvalue, the whole expression
+                                # will be replacementvalue
+                                return replacementvalue
+                            # add replacement
+                            if isinstance(np, Relational): # We only want ITE if they simplify to Relationals
+                                costsaving = measure(oldexpr) - measure(np) # ratio?
+                                if costsaving > 0:
+                                    results.append((costsaving, (i, j, np)))
+            if results:
+                # Sort results based on complexity
+                results = list(reversed(sorted(results, key=lambda pair: pair[0])))
+                # Replace the one providing most simplification
+                cost, replacement = results[0]
+                i, j, newrel = replacement
+                # Remove the old relationals
+                del Rel[j]
+                del Rel[i]
+                if newrel != ~dominatingvalue:
+                    # Insert the new one (no need to insert a value that will
+                    # not affect the result)
+                    Rel.append(newrel.canonical)
+                # We did change comething so try again
```

Comment:
comething->something

Reply:
Thanks! Will fix. (I'll wait until the improved canonical is hopefully merged as it will simplify this quite a bit.)